app/src/main/java/tech/relaycorp/gateway/data/model/MessageAddress.kt

```diff
@@ -0,0 +1,37 @@
+package tech.relaycorp.gateway.data.model
+
+sealed class MessageAddress {
+
+    val type
+        get() = when (this) {
+            is PublicMessageAddress -> Type.Public
+            is PrivateMessageAddress -> Type.Private
+        }
+    val value
+        get() = when (this) {
+            is PublicMessageAddress -> publicValue
+            is PrivateMessageAddress -> privateValue
+        }
+
+    companion object {
+        fun of(value: String) =
+            if (value.contains(":")) {
```

Comment:
It would be nice if the core library had some Address functionality for the apps to lean on. I basically just copied this class over from the Courier app.

Reply:
Agreed! I'll create a PR when https://github.com/relaycorp/relaynet-jvm/issues/62 is done